File: 001. Two Sum/001.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> nums_map;
        for (int i = 0; i < nums.size(); i++) {
            nums_map.insert(std::make_pair(nums[i], i));
        }
        for (int i = 0; i < nums.size(); i++) {
            int key = target - nums[i];
            if (nums_map.find(key) != nums_map.end() && nums_map[key] != i) {
                return{ i, nums_map[key] };
            }
        }
        return{};
    }
};

File: 001. Two Sum/001.go
func twoSum(nums []int, target int) []int {
    nums_map := make(map[int]int);
    for i := 0; i < len(nums); i++ {
        nums_map[nums[i]] = i;
    }
    for i := 0; i < len(nums); i++ {
        var key int = target - nums[i];
        if _, ok := nums_map[key]; ok && nums_map[key] != i{
            return []int {i, nums_map[key]}
        }
    }
    return []int {}
}